### PR TITLE
add iam roles to redshift

### PR DESCRIPTION
**Pull request by andharris**
Attempting to add support for iam roles to the redshift via https://github.com/spulec/moto/issues/1487

### CODE REVIEW

File: moto/redshift/models.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def __init__(self, redshift_backend, cluster_identifier, node_type, master_usern
                  preferred_maintenance_window, cluster_parameter_group_name,
                  automated_snapshot_retention_period, port, cluster_version,
                  allow_version_upgrade, number_of_nodes, publicly_accessible,
-                 encrypted, region_name, tags=None):
+                 encrypted, region_name, tags=None, iam_roles_arn=[]):
```
Comment (spulec) [inline moto/redshift/models.py]: Using mutable default arguments can be dangerous; see [here](http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments). Probably better to default to None and then set it to an empty list if None after.

File: moto/redshift/models.py [hunk 2]
```diff
@@ -351,14 +357,15 @@ class Snapshot(TaggableResourceMixin, BaseModel):
 
     resource_type = 'snapshot'
 
-    def __init__(self, cluster, snapshot_identifier, region_name, tags=None):
+    def __init__(self, cluster, snapshot_identifier, region_name, tags=None, iam_roles_arn=[]):
```
Comment (spulec) [inline moto/redshift/models.py]: Same thing here

**Comment by spulec** [comment]
Two small comments, but otherwise looks great. Excited to get this in.

File: moto/redshift/models.py (hunk 1 above)
Comment (andharris) [reply moto/redshift/models.py]: Ah yes, this definitely needs to be avoided. I've updated the default to `None`.

**Comment by spulec** [comment]
Awesome. Thanks for this and welcome to Moto!

**Comment by andharris** [comment]
Thank you. This is a great project and happy to be able to make a small contribution

**Pull request merged**